src/hooks/usePastTransaction/usePastTransaction.tsx

```diff
@@ -0,0 +1,42 @@
+import { useEffect, useState } from "react";
+import { PastTransactionsResult } from "../../types";
+import { usePrevious } from "../usePrevious";
+import { getPastTransactions } from "../../services/quota";
+import { Sentry } from "../../utils/errorTracking";
+
+export type PastTransactionHook = {
+  pastTransactionsResult: PastTransactionsResult | null;
+};
+
+export const usePastTransaction = (
+  id: string,
+  authKey: string,
+  endpoint: string
+): PastTransactionHook => {
+  const [
+    pastTransactionsResult,
+    setPastTransactionsResult
+  ] = useState<PastTransactionsResult | null>(null);
+  const prevId = usePrevious(id);
+
+  useEffect(() => {
+    const fetchPastTransactions = async (): Promise<void> => {
+      const pastTransactionsResponse = await getPastTransactions(
+        id,
+        authKey,
+        endpoint
+      );
+
+      setPastTransactionsResult(pastTransactionsResponse);
+    };
+
+    if (prevId != id)
+      fetchPastTransactions().catch(() => {
+        Sentry.captureException(`Unable to fetch past transactions: ${id}`);
+      });
```

Comment:
Are we able to move the catch into the `fetchPastTransactions` instead?
Also, what will the user see when this error occurs? Or will this be handled in the other PR on past transactions?

Reply:
Actually the catch is already done in the `fetchPastTransaction`. I added another sentry because I wasn't sure how the error should be handled and moreover there is an ongoing revamping of error msgs. Because this is called at noQuota card, even if error thrown, its likely due to backend error. So the handling of exception should be discussed and handled in a separate PR

Reply:
```js
  try {
    response = await fetchWithValidator(
      PastTransactionsResult,
      `${endpoint}/transactions/${id}`,
      {
        method: "GET",
        headers: {
          Authorization: key
        }
      }
    );
    return response;
  } catch (e) {
    if (e instanceof ValidationError) {
      Sentry.captureException(e);
    }
    throw new PastTransactionError(e.message);
  }
```

Reply:
Ok got it!